pkg/sql/codegen/xgboost/codegen.go

```diff
@@ -203,6 +204,41 @@ func resolveFeatureMeta(fds []ir.FieldDesc) ([]byte, []string, error) {
 	return f, featureNames, e
 }
 
+func generateFeatureColumnCode(fc ir.FeatureColumn) (string, error) {
+	return codegen.GenerateFeatureColumnCode(fc, "xgboost_extended")
+}
+
+func deriveFeatureColumnCode(fcs []ir.FeatureColumn, labelFc ir.FeatureColumn) (featureColumnsCode string, fieldDescs []ir.FieldDesc, label ir.FieldDesc, err error) {
```

Comment:
Feature columns and "Field Descs" are not equal, the function called `deriveFeatureColumnCode` while it's returning feature columns together with field descs.

Reply:
Done. Renamed to `deriveFeatureColumnCodeAndLabelDescs` .